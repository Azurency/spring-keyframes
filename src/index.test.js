import { spring } from './'

it('returns a keyframe array, with duplicate values removed', () => {
  const animation = spring({
    from: {
      opacity: 0,
      x: 0,
      scale: 0,
    },
    to: {
      opacity: 1,
      x: 100,
      scale: 1,
    },
  })

  expect(animation).toMatchSnapshot()
})
