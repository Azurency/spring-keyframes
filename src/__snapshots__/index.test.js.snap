// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a keyframe array, with duplicate values removed 1`] = `
Array [
  "0% {  opacity: 0.1; transform: translateX(0.1px) scale3d(0.1, 0.1, 1); }",
  "1% {  opacity: 0.1; transform: translateX(1.9px) scale3d(0.1, 0.1, 1); }",
  "2% {  opacity: 0.1; transform: translateX(5.7px) scale3d(0.1, 0.1, 1); }",
  "3% {  opacity: 0.2; transform: translateX(11.1px) scale3d(0.2, 0.2, 1); }",
  "4% {  opacity: 0.2; transform: translateX(17.7px) scale3d(0.2, 0.2, 1); }",
  "5% {  opacity: 0.3; transform: translateX(23.8px) scale3d(0.3, 0.3, 1); }",
  "6% {  opacity: 0.4; transform: translateX(31.5px) scale3d(0.4, 0.4, 1); }",
  "7% {  opacity: 0.4; transform: translateX(39.5px) scale3d(0.4, 0.4, 1); }",
  "8% {  opacity: 0.5; transform: translateX(47.6px) scale3d(0.5, 0.5, 1); }",
  "9% {  opacity: 0.6; transform: translateX(55.4px) scale3d(0.6, 0.6, 1); }",
  "10% {  opacity: 0.7; transform: translateX(61.7px) scale3d(0.7, 0.7, 1); }",
  "11% {  opacity: 0.7; transform: translateX(68.9px) scale3d(0.7, 0.7, 1); }",
  "12% {  opacity: 0.8; transform: translateX(75.5px) scale3d(0.8, 0.8, 1); }",
  "13% {  opacity: 0.9; transform: translateX(81.5px) scale3d(0.9, 0.9, 1); }",
  "14% {  opacity: 0.9; transform: translateX(87px) scale3d(0.9, 0.9, 1); }",
  "15% {  opacity: 1; transform: translateX(91px) scale3d(1, 1, 1); }",
  "16% {  opacity: 1; transform: translateX(95.2px) scale3d(1, 1, 1); }",
  "17% {  opacity: 1; transform: translateX(98.8px) scale3d(1, 1, 1); }",
  "18% {  opacity: 1.1; transform: translateX(101.8px) scale3d(1.1, 1.1, 1); }",
  "19% {  opacity: 1.1; transform: translateX(104.2px) scale3d(1.1, 1.1, 1); }",
  "20% {  opacity: 1.1; transform: translateX(105.8px) scale3d(1.1, 1.1, 1); }",
  "21% {  opacity: 1.1; transform: translateX(107.3px) scale3d(1.1, 1.1, 1); }",
  "22% {  opacity: 1.1; transform: translateX(108.3px) scale3d(1.1, 1.1, 1); }",
  "23% {  opacity: 1.1; transform: translateX(109px) scale3d(1.1, 1.1, 1); }",
  "24% {  opacity: 1.1; transform: translateX(109.3px) scale3d(1.1, 1.1, 1); }",
  "25% {  opacity: 1.1; transform: translateX(109.4px) scale3d(1.1, 1.1, 1); }",
  "26% {  opacity: 1.1; transform: translateX(109.2px) scale3d(1.1, 1.1, 1); }",
  "27% {  opacity: 1.1; transform: translateX(108.9px) scale3d(1.1, 1.1, 1); }",
  "28% {  opacity: 1.1; transform: translateX(108.4px) scale3d(1.1, 1.1, 1); }",
  "29% {  opacity: 1.1; transform: translateX(107.8px) scale3d(1.1, 1.1, 1); }",
  "30% {  opacity: 1.1; transform: translateX(107.3px) scale3d(1.1, 1.1, 1); }",
  "31% {  opacity: 1.1; transform: translateX(106.5px) scale3d(1.1, 1.1, 1); }",
  "32% {  opacity: 1.1; transform: translateX(105.8px) scale3d(1.1, 1.1, 1); }",
  "33% {  opacity: 1.1; transform: translateX(105.1px) scale3d(1.1, 1.1, 1); }",
  "34% {  opacity: 1.1; transform: translateX(104.3px) scale3d(1.1, 1.1, 1); }",
  "35% {  opacity: 1.1; transform: translateX(103.7px) scale3d(1.1, 1.1, 1); }",
  "36% {  opacity: 1.1; transform: translateX(103px) scale3d(1.1, 1.1, 1); }",
  "37% {  opacity: 1.1; transform: translateX(102.4px) scale3d(1.1, 1.1, 1); }",
  "38% {  opacity: 1.1; transform: translateX(101.9px) scale3d(1.1, 1.1, 1); }",
  "39% {  opacity: 1.1; transform: translateX(101.3px) scale3d(1.1, 1.1, 1); }",
  "40% {  opacity: 1.1; transform: translateX(101px) scale3d(1.1, 1.1, 1); }",
  "41% {  opacity: 1.1; transform: translateX(100.6px) scale3d(1.1, 1.1, 1); }",
  "42% {  opacity: 1.1; transform: translateX(100.2px) scale3d(1.1, 1.1, 1); }",
  "43% {  opacity: 1; transform: translateX(99.9px) scale3d(1, 1, 1); }",
  "44% {  opacity: 1; transform: translateX(99.7px) scale3d(1, 1, 1); }",
  "45% {  opacity: 1; transform: translateX(99.5px) scale3d(1, 1, 1); }",
  "46% {  opacity: 1; transform: translateX(99.4px) scale3d(1, 1, 1); }",
  "47% {  opacity: 1; transform: translateX(99.3px) scale3d(1, 1, 1); }",
  "48% {  opacity: 1; transform: translateX(99.2px) scale3d(1, 1, 1); }",
  "53% {  opacity: 1; transform: translateX(99.3px) scale3d(1, 1, 1); }",
  "55% {  opacity: 1; transform: translateX(99.4px) scale3d(1, 1, 1); }",
  "57% {  opacity: 1; transform: translateX(99.5px) scale3d(1, 1, 1); }",
  "58% {  opacity: 1; transform: translateX(99.6px) scale3d(1, 1, 1); }",
  "60% {  opacity: 1; transform: translateX(99.7px) scale3d(1, 1, 1); }",
  "61% {  opacity: 1; transform: translateX(99.8px) scale3d(1, 1, 1); }",
  "63% {  opacity: 1; transform: translateX(99.9px) scale3d(1, 1, 1); }",
  "65% {  opacity: 1; transform: translateX(100px) scale3d(1, 1, 1); }",
  "68% {  opacity: 1.1; transform: translateX(100.1px) scale3d(1.1, 1.1, 1); }",
  "93% {  opacity: 1; transform: translateX(100px) scale3d(1, 1, 1); }",
]
`;
